test/e2e/tsdx-lint.test.ts

```diff
@@ -30,6 +30,48 @@ describe('tsdx lint', () => {
     expect(output.stdout.includes('prettier/prettier')).toBe(true);
   });
 
+  it('should succeed linting a ts file with warnings when --max-warnings is not used', () => {
+    const testFile = `${lintDir}/file-with-lint-warnings.ts`;
+    const output = shell.exec(`node dist/index.js lint ${testFile}`);
+    expect(output.code).toBe(0);
+    expect(output.stdout.includes('@typescript-eslint/no-unused-vars')).toBe(
+      true
+    );
+  });
+
+  it('should succeed linting a ts file with fewer warnings than --max-warnings', () => {
+    const testFile = `${lintDir}/file-with-lint-warnings.ts`;
+    const output = shell.exec(
+      `node dist/index.js lint ${testFile} --max-warnings 4`
+    );
+    expect(output.code).toBe(0);
+    expect(output.stdout.includes('@typescript-eslint/no-unused-vars')).toBe(
+      true
+    );
+  });
+
+  it('should succeed linting a ts file with same number of warnings as --max-warnings', () => {
+    const testFile = `${lintDir}/file-with-lint-warnings.ts`;
+    const output = shell.exec(
+      `node dist/index.js lint ${testFile} --max-warnings 3`
+    );
+    expect(output.code).toBe(0);
+    expect(output.stdout.includes('@typescript-eslint/no-unused-vars')).toBe(
+      true
+    );
+  });
+
+  it('should fail to lint a ts file with more warnings than --max-warnings', () => {
+    const testFile = `${lintDir}/file-with-lint-warnings.ts`;
+    const output = shell.exec(
+      `node dist/index.js lint ${testFile} --max-warnings 2`
+    );
+    expect(output.code).toBe(1);
+    expect(output.stdout.includes('@typescript-eslint/no-unused-vars')).toBe(
+      true
+    );
+  });
+
```

Comment:
Let's move this under the `tsx` tests but before "`should not lint`" test so that the lint error tests are grouped together as are the lint warning tests